geospaas/base_viewer/views.py

```diff
@@ -56,55 +43,42 @@ def post(self, request, *args, **kwargs):
             self.forms[counter] = element_form_class(request.POST)
 
         self.validation_and_error_check()
-        # modify attributes based on self.form
-        return self.sorting_func(request)
+
+        # modify attributes based on the forms cleaned data
+        self.filtering_func(request)
+        return self.final_rendering(request)
 
     def validation_and_error_check(self):
         for element_forms in self.forms:  # for loop for making clean data by is_valid() method
             element_forms.is_valid()
-            for errorField in element_forms.errors: # temporary error message
+            for errorField in element_forms.errors:  # temporary error message
                 print(f"errorField")
 
-    def sorting_func(self, request):
-        self.ds = CatalogDataset.objects.all()
-        self.filtering_func()
-        return self.final_rendering(request)
-
+    def filtering_func(self, request):
+        ds = CatalogDataset.objects.all()
+        if (request.method == 'POST'):
+            for counter in range(len(self.form_class)):
+                # using the filter function of each form sequentially
+                ds = self.forms[counter].filter(ds)
+            self.ds = ds
+        elif (request.method == 'GET'):
```

Comment:
Why you don't want filtering in the GET request? With the default values, it will anyway return all datasets. Either let both get() and post() use exactly the same function without if. Or make a separate function for get() and for post(). Also, don't assign to self.ds, better to return ds as output - it is more clean.
"Explicit is better than implicit"

```python
    def get(self):
        ...
        ds = self.get_all_datasets()
        ...

    def post(self):
        ....
        ds = self.get_all_datasets()
        ds = self.filter_datasets(ds)
        ...

    def get_all_datasets(self):
        return CatalogDataset.objects.all()

    def filter_datasets(self, ds):
        for form in self.forms:
            ds = form.filter(ds)
        return ds
```

Reply:
The reason for bypassing the filtering action in the GET is that there is no need to do it. IF you execute a procedure which is not needed, it only slows your performance. So, there is no need to make the pipeline elongated for GET request. This is my reasoning as far as I know.

Please consider that we can only GET and POST request for this viewer, then this if statement for distinguishing between these two is enough and simple(not complicated). There is no need to take care of PUT request or DELETE ones. So this function with this single if can manage everything for us.





If we use `ds`, that `ds` is only works for that specific function. There we need to address them to be in the `self.ds` for the functions of the calss (here `set_dataset_context`) to use the result of previous functions in the workflow of this **class-based** view. The code won't work if we don't assign it to the `self.ds`. I have tested it based on your suggestion and it won't work.